diff --git a/src/plugins/topic_viewer/TopicViewer.cc b/src/plugins/topic_viewer/TopicViewer.cc
index 97d4e5997..c03e47d7d 100644
--- a/src/plugins/topic_viewer/TopicViewer.cc
+++ b/src/plugins/topic_viewer/TopicViewer.cc
@@ -266,12 +266,13 @@ void TopicViewer::Implementation::AddField(QStandardItem *_parentItem,
     auto messageType = msgField->message_type();
 
     if (messageType)
-      this->AddField(msgItem, msgField->name(), messageType->full_name());
+      this->AddField(msgItem, std::string(msgField->name()),
+          std::string(messageType->full_name()));
 
     else
     {
-      auto msgFieldItem = this->FactoryItem(msgField->name(),
-                                            msgField->type_name());
+      auto msgFieldItem = this->FactoryItem(std::string(msgField->name()),
+                                            std::string(msgField->type_name()));
       msgItem->appendRow(msgFieldItem);
 
       this->SetItemPath(msgFieldItem);
